addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2299,8 +2299,14 @@ export class OdooEditor extends EventTarget {
             // zws if still needed.
             const el = closestElement(insertedZws);
             const next = insertedZws.nextSibling;
+            const isLineBreak = el && el.previousElementSibling && el.previousElementSibling.tagName === 'BR';
             insertedZws.remove();
             el && fillEmpty(el);
+            if (isLineBreak) {
+                // If there was already a line-break BR just before
+                // el that was removed in fillEmpty, add it back.
+                el.before(this.document.createElement('BR'));
+            }
```

Comment:
This is confusing why does `fillEmpty` remove that `<br>` in first place?

Reply:
Is does remove `<br>` on purpose. When there an empty paragraph having <br> in it and we want to add `zws` inside it then `<br>` should get removed. See the [test](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/editor/odoo-editor/test/spec/color.test.js#L58-L77) case.

Reply:
But for the test case you mentioned isn't that an issue of `applyColor` where it adds font tag after `<br>` rather than replacing it with `<br>` [here](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js#L788).

@dmo-odoo A question let's say if I have `<p><br><font></font></p>` and I `fillEmpty(font)` why should it remove the `<br>` of the previous line? [see](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L2446C1-L2449C10)

@adsa-odoo Also there is a typo in the commit message title. It should be "extra" not "axtra".

Reply:
I don't think `fillEmpty` should remove any BR. It's meant to add BR if needed, not to remove them if not needed. I think the removal should be in another function, maybe sanitize for example, wherever the current code that removes unneeded BR is.